src/pd/pd3p2.rs

```diff
@@ -0,0 +1,476 @@
+}
+
+#[derive(Debug, Clone, PartialEq, Default, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+pub struct SourceCapabilitiesExtended {
+    /// Numeric Vendor ID (assigned by the USB-IF)
+    pub vid: u32,
+    /// Numeric Product ID (assigned by the manufacturer)
+    pub pid: u32,
+    /// Numeric Value provided by the USB-IF assigned to the product
+    pub xid: u32,
+    /// Numeric Firmware version number
+    pub fw_version: u32,
+    /// Numeric Hardware version number
+    pub hw_version: u32,
+    /// Voltage Regulation
+    pub voltage_regulation: SceVoltageRegulation,
+    /// Holdup Time
+    pub holdup_time: u32,
+    /// Compliance
+    pub compliance: SceCompliance,
+    /// Touch Current
+    pub touch_current: SceTouchCurrent,
+    /// Peak Current1
+    pub peak_current1: ScePeakCurrent,
+    /// Peak Current2
+    pub peak_current2: ScePeakCurrent,
+    /// Peak Current3
+    pub peak_current3: ScePeakCurrent,
+    /// Touch Temperature conforms to:
+    pub touch_temp: SceTouchTemp,
+    /// Source Inputs
+    pub source_inputs: SceSourceInputs,
+    /// Number of Batteries/Battery Slots
+    pub num_batteries_slots: u32,
+    /// SPR Source PDP Rating
+    pub spr_source_pdp_rating: u32,
+    /// EPR Source PDP Rating
+    pub epr_source_pdp_rating: u32,
+}
+
+/// See USPD - 6.5.3 Get_Battery_Cap Message
+#[derive(Debug, Clone, PartialEq, Default, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+pub struct BatteryCapData {
+    #[c_api(no_prefix)]
+    pub batteries_fixed: [u32; 4],
+    #[c_api(no_prefix)]
+    pub batteries_hotswappable: [u32; 4],
+}
+
+#[derive(Debug, Clone, PartialEq, Default, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+pub struct BSDBatteryInfo {
+    /// Invalid Battery reference
+    pub invalid_battery_reference: bool,
+    /// Battery is present when set
+    pub battery_present: bool,
+    /// Battery is Charging.
+    pub battery_charging: bool,
+    /// Battery is Discharging.
+    pub battery_discharging: bool,
+    /// Battery is Idle.
+    pub battery_idle: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Default, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+pub struct BatteryStatusData {
+    /// Battery’s State of Charge (SoC) in 0.1 WH increments
+    /// Note: 0xFFFF = Battery’s SOC unknown
+    pub battery_present_capacity: u32,
+    /// Battery Info
+    pub battery_info: BSDBatteryInfo,
+}
+
+#[derive(Debug, Clone, PartialEq, Default, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+pub struct RevisionMessageData {
+    /// Revision.major
+    pub revision_major: u32,
+    /// Revision.minor
+    pub revision_minor: u32,
+    /// Version.major
+    pub version_major: u32,
+    /// Version.minor
+    pub version_minor: u32,
+    /// Reserved, Shall be set to zero
+    pub reserved: u32,
+}
+
+#[derive(Debug, Clone, PartialEq, Default, N, Copy, CApiWrapper)]
+#[c_api(prefix = "Pd3p2", repr_c)]
+/// See USB PD 3.2 - Table 6.17 “Fixed Supply PDO – Sink”
```

Comment:
Looks like an incorrect copy&paste reference

Reply:
Nope, they're described in the same table